esphome/components/hm3301/caqi_calculator.h

```diff
@@ -37,7 +37,7 @@ class CAQICalculator : public AbstractAQICalculator {
     int conc_lo = array[grid_index][0];
     int conc_hi = array[grid_index][1];
 
-    int aqi = ((aqi_hi - aqi_lo) / (conc_hi - conc_lo)) * (value - conc_lo) + aqi_lo;
+    int aqi = ((aqi_hi - aqi_lo) / (double) (conc_hi - conc_lo)) * (value - conc_lo) + aqi_lo;
```

Comment:
We use single-precision floats for performance reasons:
```suggestion
    int aqi = ((aqi_hi - aqi_lo) / (float) (conc_hi - conc_lo)) * (value - conc_lo) + aqi_lo;
```

Reply:
Actually, given that the end result is converted to an integer anyway, rearranging the order should also fix the calculation without incurring the performance cost of floats:
```suggestion
    int aqi = (value - conc_lo) * (aqi_hi - aqi_lo) / (conc_hi - conc_lo) + aqi_lo;
```

Reply:
Thanks, fixed